pkg/utils/utils.go

```diff
@@ -309,6 +303,15 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 						return err
 					}
 				}
+
+				// only set GUID and MAC for VF with default driver
+				if strings.EqualFold(iface.LinkType, "IB") {
+					if err = setVfGuid(addr, pfLink); err != nil {
+						return err
+					}
+				} else if err = setVfAdminMac(addr, pfLink); err != nil {
```

Comment:
I think we can still set VF admin mac w/ vfio-pci driver, right?
Just like sriov-cni can set VF attributes(admin-mac, vlan, trust, spoofchk etc) in vfio-pci mode.

Reply:
Does it make any sense to set the VF MAC when using vfio-pci driver? We invoke `vfIsReady` in `setVfAdminMac`, it returns error with vfio-pci driver.

Reply:
This is to resolve https://github.com/k8snetworkplumbingwg/sriov-network-operator/issues/244

Reply:
I am afraid this will result in adapters with vfio-pci to use the ff:ff:ff:ff:ff:ff:ff address. Can we just skip the `setVfAdminMac` in case we did not change the vfNum? Considering this agent always creates the VFs, the mac will always be set, because it would happen when we created them.

Reply:
> Does it make any sense to set the VF MAC when using vfio-pci driver? We invoke `vfIsReady` in `setVfAdminMac`, it returns error with vfio-pci driver.

I think yes. For some vendor NICs, the default admin MAC is all-zero, which means all packet (including packets with spoofed MAC) can be sent by app using the VF (when VF spoofChk is set to on).

Reply:
Fixed. PTAL.